docs/xmllib-docs/overview.md

```diff
@@ -0,0 +1,252 @@
+    date_range_separator="-",
+    date_format=xmllib.DateFormat.DD_MM_YYYY,
+)
+
+resource = resource.add_date(prop_name=":hasDate", value=formatted_date)
+```
+
+Find further functions to convert values [here](value-converters.md)
+
+
+### Data Cleaning Functions
+
+Often times we require similar functions to process and clean the data before we can add them to the resource.
+
+For example many strings extracted from the data source, represent not one value but a list of values.
+We created the following function to clean your input.
+
+```python
+input_string = "This should be\na list\nof strings."
+cleaned_list = xmllib.create_list_from_input(
+    input_value=input_string,
+    separator="\n"
+)
+```
+
+You can find this and other functions to help you process your data [here](general-functions.md).
+
+Please contact us with a feature request if you cannot find a function in the list, 
+but you feel that it may have generic applications and would be helpful to other users.
+
+### Adding A File To a Resource
+
+In the ontology you can 
+[specify](../data-model/json-project/ontologies.md#resource-super) that a resource should have a file attached.
+
+You can add all types of files to the resource with the following function.
+Please note that only one file or IIIF-URI (see below) are allowed per resource.
+
+```python
+image_resource = image_resource.add_file(
+    filename="cat.tiff",
+    license=xmllib.LicenseRecommended.CC.BY,
+    copyright_holder="Mouse University",
+    authorship=["Minnie Mouse"],
+)
+```
+
+We require that legal information be provided for resources with files. 
+While the copyright holder and authorships are free text fields.
+All licenses referenced must be predefined, [look here for our recommended options](./licenses/recommended.md).
+
+Please contact us if you cannot find the required license in either our recommended or [other options](./licenses/other.md).
+
+### Adding IIIF-URIs To a Resource
+
+We provide the option to add a IIIF-URI to a Resource with the super-class `StillImageRepresentation`.
+This way you can reference images that lie on other servers.
+
+```python
+resource = resource.add_iiif_uri(
+    iiif_uri="https://iiif.wellcomecollection.org/image/b20432033_B0008608.JP2/full/1338%2C/0/default.jpg",
+    license=xmllib.LicenseRecommended.CC.BY_NC,
+    copyright_holder="Wellcome Collection",
+    authorship=["Cavanagh, Annie"]
+)
+```
+
+Please note that the IIIF-URI must follow the official syntax specified [here](https://iiif.io/api/image/3.0/#2-uri-syntax).
+
+### Adding a Comment To a Value
+
+It is possible to add a comment to individual values or files with the following parameter.
+
+```python
+resource = resource.add_integer_optional(
+    prop_name=":hasInt",
+    value=1,
+    comment="This text is a comment on this integer.",
+)
+```
+
+Please note, that if you add a comment to values with the add multiple functions, 
+all the values will have the same comment.
```

Comment:
```suggestion
Please note that if you add a comment to values with the `add_xyz_multiple()` methods, 
all values will have the same comment.
```

Reply:
This is already quite specific, so I think this remark should be removed altogether.